petastorm/tests/test_spark_dataset_converter.py

```diff
@@ -234,6 +234,20 @@ def test_df_caching(test_ctx):
     assert converter12.cache_dir_url != converter22.cache_dir_url
 
 
+def test_df_delete_caching_meta(test_ctx):
+    from petastorm.spark.spark_dataset_converter import _cache_df_meta_list
+    df1 = test_ctx.spark.range(10)
+    df2 = test_ctx.spark.range(20)
+    converter1 = make_spark_converter(df1)
+    converter2 = make_spark_converter(df2)
+    converter1.delete()
```

Comment:
I think we letting `make_tf_dataset` raise 'dir not exist' error will be good enough.
It is hard to make an ideal destruction, because multiple converters may share the same cache dir, destructing all of them makes code complex.